h2o-core/src/main/java/water/api/DatasetServlet.java

```diff
@@ -32,6 +32,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) {
           .setEscapeQuotes(Boolean.parseBoolean(escape_quotes_string));
       InputStream is = dataset.toCSV(parms);
       response.setContentType("application/octet-stream");
+      // todo: get this somehow from dataset
```

Comment:
I think H2O parser always convert data to `utf-8`: needs confirmation though.

Reply:
it may be more complicated than that: https://docs.h2o.ai/h2o/latest-stable/h2o-docs/getting-data-into-h2o.html?highlight=utf

this is true for CSV/arff imports, bot for example, hard to know what happens with JDBC imports…

Reply:
thanks!

Reply:
most extended ascii encodings (except the always annoying windows-1252 probably) should convert nicely to `utf-8`, the main issue is with `utf-16` and windows then: and it's not clear to  me if H2O supports those encodings today.

Reply:
I've found this check: water.parser.ParseSetup#checkEncoding, from this it seems like UTF16 is not supported

Reply:
It's not supported from parsers, correct. But we have other types of import, for example jdbc (direct import from RDBS, or also for Hive tables) or Hive import from hdfs who don't require any parsing if I'm correct…

Reply:
Apparently the Hive import calls `guessSetup` so it will detect and reject `utf-16`.
Then even if it's not checked in `SQLManager`, I guess we can say that H2O doesn't support `utf-16`.